tests/unit/nupic/algorithms/sdr_classifier_test.py

```diff
@@ -0,0 +1,445 @@
+    retval = []
+    for recordNum in xrange(10):
+      retval = self._compute(classifier, recordNum, [1, 5], 0, 10)
+
+    self.assertEqual(retval['actualValues'][0], 10)
+    self.assertGreater(retval[1][0], 0.9)
+
+
+  def testSingleValue0Steps(self):
+    """Send same value 10 times and expect high likelihood for prediction
+    using 0-step ahead prediction"""
+    classifier = self._classifier(steps=[0], alpha=1.0)
+
+    # Enough times to perform Inference and learn associations
+    retval = []
+    for recordNum in xrange(10):
+      retval = self._compute(classifier, recordNum, [1, 5], 0, 10)
+
+    self.assertEqual(retval['actualValues'][0], 10)
+    self.assertGreater(retval[0][0], 0.9)
+
+
+  def testComputeResultTypes(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    result = c.compute(recordNum=0,
+                       patternNZ=[1, 5, 9],
+                       classification= {'bucketIdx': 4, 'actValue': 34.7},
+                       learn=True,
+                       infer=True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(type(result['actualValues']), list)
+    self.assertEqual(len(result['actualValues']), 1)
+    self.assertEqual(type(result['actualValues'][0]), float)
+    self.assertEqual(type(result[1]), numpy.ndarray)
+    self.assertEqual(result[1].itemsize, 8)
+    self.assertAlmostEqual(result['actualValues'][0], 34.7, places=5)
+
+
+  def testBucketIdxNumpyInt64Input(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    result = c.compute(0, [1, 5, 9],
+                       {'bucketIdx': numpy.int64(4), 'actValue': 34.7}, True,
+                       True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(len(result['actualValues']), 1)
+    self.assertAlmostEqual(result['actualValues'][0], 34.7, places=5)
+
+
+  def testCompute1(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    result = c.compute(recordNum=0,
+                       patternNZ=[1, 5, 9],
+                       classification={'bucketIdx': 4, 'actValue': 34.7},
+                       learn=True,
+                       infer=True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(len(result['actualValues']), 1)
+    self.assertAlmostEqual(result['actualValues'][0], 34.7, places=5)
+
+
+  def testCompute2(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    c.compute(recordNum=0, patternNZ=[1, 5, 9],
+              classification={'bucketIdx': 4, 'actValue': 34.7},
+              learn=True, infer=True)
+    result = c.compute(recordNum=1, patternNZ=[1, 5, 9],
+                       classification={'bucketIdx': 4, 'actValue': 34.7},
+                       learn=True, infer=True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertAlmostEqual(result['actualValues'][4], 34.7, places=5)
+
+
+  def testComputeComplex(self):
```

Comment:
We don't see to have test cases where learn=False and infer=True, or vice versa.  Also, if both are False, an exception should be raised.